framework/source/class/qx/bom/Notification.js

```diff
@@ -0,0 +1,305 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2004-2014 1&1 Internet AG, Germany, http://www.1und1.de
+
+   Authors:
+     * Cajus Pollmeier (cajus)
+
+************************************************************************ */
+
+/**
+ * This class offers a constant API over the Notification Spec:
+ * http://www.w3.org/TR/notifications/
+ *
+ * It forwards all the browsers support if supported.
+ *
+ * *Example*
+ *
+ * <pre class="javascript">
+ * var notifications = qx.bom.Notification.getInstance();
+ *
+ * var button = new qx.ui.form.Button("Notify me!");
+ * button.addListener("execute", function() {
+ *   notifications.show("Information", "Hey there!", "icon/64/status/dialog-information.png", 5000);
+ * });
+ *
+ * // Enable button if supported
+ * button.setEnabled(qx.core.Environment.get("html.notification"));
+ *
+ * </pre>
+ *
+ * **Note**
+ *
+ * A notification can only be send if the user allows these notifications to
+ * be shown. <code>qx.bom.Notification</code> automatically tries to
+ * trigger a browser dialog which asks the user for permission.
+ *
+ * But there is a restriction: the dialog will only show up,
+ * if the initial request was triggered from within an
+ * interactive event handler. So there is most likely a mouse or key
+ * event involved initially to trigger it properly.
+ *
```

Comment:
maybe: 

But there is a restriction: the dialog will only show up if it is triggered by code that is running inside a request handler for an interactive browser event like a mouse click or a keyboard interaction.